fw/http_msg.c

```diff
@@ -553,13 +553,20 @@ tfw_http_msg_hdr_close(TfwHttpMsg *hm)
 	TfwHttpHdrTbl *ht = hm->h_tbl;
 	TfwHttpParser *parser = &hm->stream->parser;
 	unsigned int id = parser->_hdr_tag;
+	bool is_srv_conn = TFW_CONN_TYPE(hm->conn) & Conn_Srv;
 
 	BUG_ON(parser->hdr.flags & TFW_STR_DUPLICATE);
 	BUG_ON(id > TFW_HTTP_HDR_RAW);
 
 	/* Close just parsed header. */
 	parser->hdr.flags |= TFW_STR_COMPLETE;
 
+	if (!is_srv_conn &&
+	    (!tfw_http_check_header_size(hm) ||
+	     !tfw_http_check_headers_count(hm) ||
+	     !tfw_http_check_header_list_size(hm)))
```

Comment:
Also I want to pay your attention, for http2 we can get header size even before decoding header thanks to hpack format.

Reply:
I move this check to hpack for HTTP2, but i we have several problems here:
- In case of huffman encoding hp->length != header length and we should decode header, also in case of TCP segmentation this checks will be executed on every data segment. Because of this problems I add them after decoding, so I don't think that we will gain a lot from this change.